tst/EnergyPlus/unit/HeatBalanceKivaManager.unit.cc

```diff
@@ -260,6 +260,299 @@ TEST_F(EnergyPlusFixture, HeatBalanceKiva_SetInitialBCs)
+        "  Temperature;              !- Unit Type",
+        " ",
+        "ScheduleTypeLimits,",
+        "  Control Type,             !- Name",
+        "  0,                        !- Lower Limit Value",
+        "  4,                        !- Upper Limit Value",
+        "  Discrete,                 !- Numeric Type",
+        "  Dimensionless;            !- Unit Type",
+        " ",
+        "ScheduleTypeLimits,",
+        "  On/Off,                   !- Name",
+        "  0,                        !- Lower Limit Value",
+        "  1,                        !- Upper Limit Value",
+        "  Discrete,                 !- Numeric Type",
+        "  Dimensionless;            !- Unit Type",
+        " ",
+        "ScheduleTypeLimits,",
+        "  Any Number,               !- Name",
+        "  ,                         !- Lower Limit Value",
+        "  ,                         !- Upper Limit Value",
+        "  Continuous;               !- Numeric Type",
+        " ",
+        "ScheduleTypeLimits,",
+        "  Velocity,                 !- Name",
+        "  ,                         !- Lower Limit Value",
+        "  ,                         !- Upper Limit Value",
+        "  Continuous,               !- Numeric Type",
+        "  Velocity;                 !- Unit Type",
+        " ",
+        "ScheduleTypeLimits,",
+        "  Activity Level,           !- Name",
+        "  0,                        !- Lower Limit Value",
+        "  ,                         !- Upper Limit Value",
+        "  Continuous,               !- Numeric Type",
+        "  ActivityLevel;            !- Unit Type",
+        " ",
+        "ScheduleTypeLimits,",
+        "  Dimensionless,            !- Name",
+        "  -1,                       !- Lower Limit Value",
+        "  1,                        !- Upper Limit Value",
+        "  Continuous;               !- Numeric Type",
+    });
+
+    ASSERT_TRUE(process_idf(idf_objects));
+
+    bool ErrorsFound(false); // If errors detected in input
+    HeatBalanceManager::GetZoneData(*state, ErrorsFound);
+    ASSERT_FALSE(ErrorsFound);
+
+    state->dataEnvrn->DayOfYear_Schedule = 1;      // must initialize this to get schedules initialized
+    state->dataEnvrn->DayOfWeek = 1;               // must initialize this to get schedules initialized
+    state->dataGlobal->HourOfDay = 1;              // must initialize this to get schedules initialized
+    state->dataGlobal->TimeStep = 1;               // must initialize this to get schedules initialized
+    state->dataGlobal->NumOfTimeStepInHour = 1;    // must initialize this to get schedules initialized
+    state->dataGlobal->MinutesPerTimeStep = 60;    // must initialize this to get schedules initialized
+    ScheduleManager::ProcessScheduleInput(*state); // read schedules
+
+    state->files.inputWeatherFilePath.filePath = configured_source_directory() / "tst/EnergyPlus/unit/Resources/HeatBalanceKivaManagerOSkyTest.epw";
+    km.setupKivaInstances(*state);
```

Comment:
So before, this unit test would have has error output?

Reply:
Before this test just would have crashed, so if there's a better way to verify this type of failure, I'm happy to change it.

Reply:
No that's fine, that's what I figured.  Thanks.  If you have any comment on the calling order note that would be great to hear so we can wrap up this little fix.

Reply:
If the test throws an exception, you could use `ASSERT_ANY_THROW`. If the test will hard crash the program, you could use `ASSERT_DEATH` or `ASSERT_EXIT` depending on what the program does. Or structure the test to not throw or crash.